### PR TITLE
Automatically create output folder if required (2)

### PR DESCRIPTION
**Pull request by BoatsAreRockable**
@le717 I think this is finally sorted now. It's amazing how much trouble such a small amount of code has caused me!

### CODE REVIEW

**Comment by le717** [comment]
I'll take a look ASAP, hopefully after work today.

File: app/js/main.js [hunk 1]
```diff
@@ -658,7 +675,23 @@ function _getDefaultDirectory() {
 }
 
 /**
-* COnverting frames to png
+ * Create the default save directory if needed
```
Comment (le717) [inline app/js/main.js]: Nit (for future reference): all function descriptions should end with a period.

File: app/js/main.js [hunk 2]
```diff
@@ -658,7 +675,23 @@ function _getDefaultDirectory() {
...
+ */
+function createSaveDirectory() {
+    "use strict";
+    mkdirp(frameExportDirectory, function (err) {
+        if (err) {
+            console.error(`Failed to create directory at ${frameExportDirectory}`);
+            notifyError(`${frameExportDirectory} does not exist. Failed to create a directory at this location.`);
```
Comment (le717) [inline app/js/main.js]: Add `console.error(err);` before the call to `notifyInfo()` so we can know exactly what happened.

File: app/js/main.js [hunk 3]
```diff
@@ -725,9 +758,11 @@ function _writeFile(file, data) {
     "use strict";
      fs.writeFile(file, data, function(err) {
         if (err) {
-            throw err;
+            console.error(`Failed to save ${file} to disk`);
```
Comment (le717) [inline app/js/main.js]: Again, add `console.error(err);` before calling `createSaveDirectory()` so we can get details.

File: app/js/main.js [hunk 4]
```diff
@@ -725,9 +758,11 @@ function _writeFile(file, data) {
...
+            createSaveDirectory();
```
Comment (le717) [inline app/js/main.js]: I'm not so sure about this call. It's assuming the error was caused by a missing output folder, which is not always the case. We also talked about the unlikelyhood of the output folder being ripped out from under the app.

I'll leave it up to do. Do what you think is best.

**Comment by le717** [comment]
Aside from my comments, looking good! :+1:

**Comment by le717** [comment]
BTW, you should usually delete branches once the PR has been merged. In this case, you can delete `issue-27` and `issue-27-new`. ;)

**Comment by BoatsAreRockable** [comment]
Alright, I will do that now :)

**Pull request merged**